src/Hosting/Hosting/src/Internal/WebHostOptions.cs

```diff
@@ -11,19 +11,14 @@ namespace Microsoft.AspNetCore.Hosting
 {
     internal class WebHostOptions
     {
-        public WebHostOptions() { }
-
-        public WebHostOptions(IConfiguration configuration)
-            : this(configuration, string.Empty) { }
-
-        public WebHostOptions(IConfiguration configuration, string applicationNameFallback)
+        public WebHostOptions(IConfiguration configuration, string? applicationNameFallback)
         {
             if (configuration == null)
             {
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            ApplicationName = configuration[WebHostDefaults.ApplicationKey] ?? applicationNameFallback;
+            ApplicationName = configuration[WebHostDefaults.ApplicationKey] ?? applicationNameFallback ?? string.Empty; // TODO - What should happen if there is no app name?
```

Comment:
Could this throw? I'm not sure what the sequence of these steps are but one would expect this is never un-assigned since other parts of the framework depend on it's correctness.

Reply:
Let's push this question back to the caller.
https://github.com/dotnet/aspnetcore/blob/ec2c7b478769c68870ec860592be27cc8920519b/src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs#L125
https://github.com/dotnet/aspnetcore/blob/d7187b01e87761c778a5c406802e8bfd9451f3cf/src/Hosting/Hosting/src/WebHostBuilder.cs#L223

Do we realistically expect those to return null?